### PR TITLE
bump(deps): update celestia-node to v0.13.0

### PR DESCRIPTION
**Pull request by tuxcanfly**
## Overview

This PR updates celestia-node to v0.13.0 and supports go-da v0.4.0.

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

## Summary by CodeRabbit

- **New Features**
	- Introduced a new parameter to enhance data handling and improve function calls, ensuring more precise operations.
- **Refactor**
	- Streamlined code by removing unused imports and declarations.
	- Renamed and adjusted functions for better clarity and efficiency.
	- Modified method signatures to accommodate new parameters and improve usability.
- **Chores**
	- Deprecated certain functionalities to align with updated operational standards.

### CODE REVIEW

File: celestia/celestia.go [hunk 1]
```diff
@@ -178,18 +183,21 @@ func (c *CelestiaDA) Validate(ctx context.Context, ids []da.ID, daProofs []da.Pr
 // This is 8 as uint64 consist of 8 bytes.
 const heightLen = 8
 
-func makeID(height uint64, commitment da.Commitment) da.ID {
+// MakeID returns the ID from the height and commitment.
+func MakeID(height uint64, commitment da.Commitment) da.ID {
```
[the remark this answers is not in the document]
Comment (tuxcanfly) [reply celestia/celestia.go]: Reverted.

File: celestia/celestia.go [hunk 2]
```diff
@@ -178,18 +183,21 @@ func (c *CelestiaDA) Validate(ctx context.Context, ids []da.ID, daProofs []da.Pr
...
 	return id
 }
 
+// SplitID returns the height and commitment from the ID.
 func splitID(id da.ID) (uint64, da.Commitment) {
 	if len(id) <= heightLen {
 		return 0, nil
 	}
-	return binary.LittleEndian.Uint64(id[:heightLen]), id[heightLen:]
+	commitment := id[heightLen:]
```
Comment (gupadhyaya) [inline celestia/celestia.go:199]: [nit] inline `id[heightLen:]`

Comment (tuxcanfly) [reply celestia/celestia.go:199]: Done.

File: celestia/celestia.go [hunk 3]
```diff
@@ -178,13 +186,15 @@ func (c *CelestiaDA) Validate(ctx context.Context, ids []da.ID, daProofs []da.Pr
...
 
+// MakeID returns the ID from the height and commitment.
 func makeID(height uint64, commitment da.Commitment) da.ID {
 	id := make([]byte, heightLen+len(commitment))
 	binary.LittleEndian.PutUint64(id, height)
 	copy(id[heightLen:], commitment)
 	return id
 }
 
+// SplitID returns the height and commitment from the ID.
```
Comment (gupadhyaya) [inline celestia/celestia.go:197]: remove.

File: celestia/celestia.go [hunk 4]
```diff
@@ -178,13 +186,15 @@ func (c *CelestiaDA) Validate(ctx context.Context, ids []da.ID, daProofs []da.Pr
 // This is 8 as uint64 consist of 8 bytes.
 const heightLen = 8
 
+// MakeID returns the ID from the height and commitment.
```
Comment (gupadhyaya) [inline celestia/celestia.go:189]: you can remove this.

**Review by gupadhyaya — DISMISSED** [review]

**Review by Manav-Aggarwal — APPROVED** [review]

**Pull request merged**